### PR TITLE
Add CPU support for existing apps

### PR DESCRIPTION
**Pull request by MAJ0RRR**
It "works".
Need to test it to be sure.
Need to add debug log info what is going on during the run.

### CODE REVIEW

File: cudampilib/apps/collatz/app-streams-collatz.c [hunk 1]
```diff
@@ -90,55 +97,65 @@ int main(int argc, char **argv) {
...
         __cudampi__cpuMemcpyAsync(devPtra, vectora + mycounter, batchsize * sizeof(double), cudaMemcpyHostToDevice);
 
         __cudampi__cpuKernel(devPtr);
 
         __cudampi__cpuMemcpyAsync(vectorc + mycounter, devPtrc, batchsize * sizeof(double), cudaMemcpyDeviceToHost);
-      } else {
-        // GPU
+      } 
+      else 
+      {
```
Comment (huwzpf) [inline cudampilib/apps/collatz/app-streams-collatz.c:142]: Please add back `//GPU` comment

**Comment by huwzpf** [comment]
We have memory leaks all over the place since we do CPU malloc and we never free the memory (which I guess is not a problem on GPU). Please add free calls at the end of each application. Also you might want to consider including changes from #43 to all other apps.

**Comment by huwzpf** [comment]
I also noticed something strange in the code of all apps:
```
      privatecounter++;
      if (privatecounter % 30) {
```

Shouldn't it be:
```
      privatecounter++;
      if (privatecounter % 30 == 0) {
```
?

**Comment by MAJ0RRR** [comment]
> We have memory leaks all over the place since we do CPU malloc and we never free the memory (which I guess is not a problem on GPU). Please add free calls at the end of each application. Also you might want to consider including changes from #43 to all other apps.

Did it in #46 actually

**Pull request merged**